src/app/geocms/geonode/qgsgeonodenewconnection.cpp

```diff
@@ -0,0 +1,272 @@
+/***************************************************************************
```

Comment:
All this class looks very much like `QgsNewHttpConnection` can you extend that class instead of creating a new one?

Reply:
Good suggestion thanks - lets take a look if we can do that...

Reply:
@elpaso - done. There certainly was a lot of duplicate code there!

@timlinux I noticed when doing this that none of the WFS/WMS specific settings (all those options like 'invert axis orientation','wfs version', etc) are actually utilised when loading geonode layers. They were being stored but only used when reopening this dialog. Accordingly I've removed them from the geonode server connection settings dialog (since it made this change must easier!).

My question is - should these even be required here? In my view they are all settings designed to handle specific quirks of different WFS/WMS server implementations. Given that we know in this case the connection is to a geonode server, couldn't we avoid showing these options and instead just hardcode whatever options are required for optimal performance with geonode instances?

It's a lot less options for users to mess with, and in my opinion dropping these options actually increases the user perception that QGIS/GeoNode are tightly integrated.

Reply:
@nyalldawson I've a limited knowledge of geonode, but AFAIK geonode supports different backends (GeoServer and QGIS Server, if I'm not wrong), so, if geonode just proxies the requests to WFS/WMS providers,  it might makes sense to keep the options.

Reply:
@elpaso - right. I've fixed this now, so the settings are exposed again and respected when adding layers via the source select dialog. I've tightened up some of the code surrounding this and added some tests too.

However... in the process I found a bigger issue! (Yay! Coding! Gotta love it!) Turns out these service specific settings are ignored for ALL providers when adding layers from the browser. (e.g. adding a WMS layer doesn't respect the axis/dpi/referer/etc settings, which are correctly respected when adding the same from the source select dialog).

That bug's WAY out of scope here though, so I've run away from it.

Reply:
@nyalldawson 
> That bug's WAY out of scope here though, so I've run away from it.
I'd recommend that you file a ticked for that issue, maybe we can fix it in the next pre-release bugfix round.

Reply:
Done- https://issues.qgis.org/issues/17123

Reply:
Thanks @nyalldawson for picking up on that. 👍 to deal with it in a separate ticket.